beacon-chain/sync/initial-sync/service.go

```diff
@@ -212,6 +207,26 @@ func (s *Service) Start() {
 	s.markSynced()
 }
 
+// fetchOriginSidecars fetches origin sidecars
+func (s *Service) fetchOriginSidecars(peers []peer.ID) error {
+	headSlot := s.cfg.Chain.HeadSlot()
+	headEpoch := slots.ToEpoch(headSlot)
+
+	if headEpoch >= params.BeaconConfig().FuluForkEpoch {
+		if err := s.fetchOriginColumns(peers); err != nil {
+			return errors.Wrap(err, "fetch origin columns")
+		}
+	}
+
+	if headEpoch >= params.BeaconConfig().DenebForkEpoch {
```

Comment:
What will happen is we fetch both blobs + columns at fulu. Since fulu is always > deneb

Reply:
Fixed in https://github.com/OffchainLabs/prysm/pull/15154/commits/e7ae287882fc36242a93c5760086d26208bc128b.